File: data/multilang_comparison/program_julia/src/func30.jl
include("program_julia_head.jl")
function func30(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      array16 = nothing
      if pCounter > 1
          array16 = vars.data[pCounter]
          pCounter -= 1
          array16.refC += 1
      else
          array16 = JArray(zeros(Int, 84), 84, 1, 16)
      end
      loopLimit8 = (50)/4 + 1
      for i in 1:loopLimit8 
         for i in 0:(array16.size-1) 
            array16.data[i+1] -= 1
         end
         for i in 1:array16.size
             if array16.data[i] == 5
                     return array16
            end
         end
      end
      params0 = JArray_param{Int}()
      params0.size = 1
      push!(params0.data, array16)
      params0.size += 1
      array17 = func42(params0, get_path(), loopsFactor);
      array17.refC -= 1
      if array17.refC == 0
          empty!(array17.data)
          array17 = nothing
      end
      array16.refC -= 1
      if array16.refC == 0
          empty!(array16.data)
          array16 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array24 = func34(params0, loopsFactor);
      array24.refC -= 1
      if array24.refC == 0
          empty!(array24.data)
          array24 = nothing
      end
   end
   array26 = nothing
   if pCounter > 1
       array26 = vars.data[pCounter]
       pCounter -= 1
       array26.refC += 1
   else
       array26 = JArray(zeros(Int, 795), 795, 1, 26)
   end
   return array26;
end

File: data/multilang_comparison/program_cpp_array/src/func30.cpp
#include "program_cpp_array.hpp" 
Array* func30(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      Array* array16;
      if (pCounter > 0) {
         array16 = vars->data[--pCounter];
         array16->refC++;
         DEBUG_COPY(array16->id);
      } else {
         array16 = new Array();
         array16->size = 84;
         array16->refC = 1;
         array16->id = 16;
         array16->data = new unsigned int[array16->size]();
         DEBUG_NEW(array16->id);
      }
      unsigned int loop8 = 0;
      unsigned int loopLimit8 = (50)/4 + 1;
      for(; loop8 < loopLimit8; loop8++) {
         for (int i = 0; i < array16->size; i++) {
             array16->data[i]--;
         }
         for (int i = 0; i < array16->size; i++) {
             if (array16->data[i] == 5) {
                 return array16;
             }
         }
      }
      Array_param params0;
      params0.size = 1;
      Array* data_params0[1];
      params0.data = data_params0;
      params0.data[0] = array16;
      Array* array17 = func42(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array17->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array17->refC--;
      if(array17->refC == 0) {
         delete[] array17->data;
         array17->data = nullptr;
         DEBUG_FREE(array17->id);
         delete array17;
      }
      array16->refC--;
      if(array16->refC == 0) {
         delete[] array16->data;
         array16->data = nullptr;
         DEBUG_FREE(array16->id);
         delete array16;
      }
   }
   else {
      Array_param params0;
      params0.size = 0;
      Array* data_params0[0];
      params0.data = data_params0;
      Array* array24 = func34(&params0, loopsFactor);
      DEBUG_RETURN(array24->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array24->refC--;
      if(array24->refC == 0) {
         delete[] array24->data;
         array24->data = nullptr;
         DEBUG_FREE(array24->id);
         delete array24;
      }
   }
   Array* array26;
   if (pCounter > 0) {
      array26 = vars->data[--pCounter];
      array26->refC++;
      DEBUG_COPY(array26->id);
   } else {
      array26 = new Array();
      array26->size = 795;
      array26->refC = 1;
      array26->id = 26;
      array26->data = new unsigned int[array26->size]();
      DEBUG_NEW(array26->id);
   }
   return array26;
}

